holoviews/plotting/mpl/chart.py

```diff
@@ -1258,3 +1258,47 @@ class SideSpikesPlot(AdjoinedPlot, SpikesPlot):
         all axis labels including ticks and ylabel. Valid options are 'left',
         'right', 'bare' 'left-bare' and 'right-bare'.""")
 
+
+
+class SegmentPlot(ColorbarPlot):
+    """
+    Segments are lines in 2D space where each two key dimensions specify a
+    (x, y) node of the line.
+    """
+    style_opts = PathPlot.style_opts + ['cmap']
+
+    _nonvectorized_styles = ['cmap']
+
+    _plot_methods = dict(single='segment')
+
+    def init_artists(self, ax, plot_args, plot_kwargs):
+        if 'c' in plot_kwargs:
+            plot_kwargs['array'] = plot_kwargs.pop('c')
+        if 'vmin' in plot_kwargs and 'vmax' in plot_kwargs:
+            plot_kwargs['clim'] = plot_kwargs.pop('vmin'), plot_kwargs.pop('vmax')
+        line_segments = LineCollection(*plot_args, **plot_kwargs)
+        ax.add_collection(line_segments)
+        return {'artist': line_segments}
+
+    def get_data(self, element, ranges, style):
+        # Get [x0, y0, x1, y1, ...]
+        x0idx, y0idx, x1idx, y1idx = (
+            (1, 0, 3, 2) if self.invert_axes else (0, 1, 2, 3)
+        )
+        # Compute segments
+        dims = element.dimensions()
+        data = [[(x0, y0), (x1, y1)] for x0, y0, x1, y1
+                in element.array([x0idx, y0idx, x1idx, y1idx])]
+
+        with abbreviated_exception():
+            style = self._apply_transforms(element, ranges, style)
+        return (data,), style, {'dimensions': dims}
+
+    def get_extents(self, element, ranges, range_type='combined'):
+        kdims = element.kdims
+        for kdim in [kdims[i].name for i in range(2)]:
+            new_range = {}
+            for r in ranges[kdim]:
+                new_range[r] = max_range([ranges[kd.name][r] for kd in kdims])
+            ranges[kdim] = new_range
+        return super(SegmentPlot, self).get_extents(element, ranges, range_type)
```

Comment:
May have to implement this twice, we still don't have a good solution for element specific code that's shared across backends.

Reply:
In the long term if we want to add more geometry elements we should consider whether it'd be worth it to add "virtual" dimensions to elements which can return the x/y/(z) ranges rather than just the ranges of each column.

Reply:
You mean port `get_extents()` to the bokeh backend?

Reply:
Right.

Reply:
So in fixing get_extent (turned out my implementation didn't do much), I found that in the bokeh backend, get_extent is being called twice for a simple Segments element - in mpl only once. Out of curiosity, any idea why?

Reply:
I think in bokeh it calls it during initialization and then calls it again in `_update_plot`. Since the ranges are all cached that shouldn't be very expensive but we should probably look into fixing it at some point.